script.js

```diff
@@ -56,7 +59,29 @@ searchBar.addEventListener("keyup", function(p){
   })
   makePageForEpisodes(filteredList);
 });
- 
-  
+ //display the Episodes on the selector menu
+ const myList = document.getElementById("list");
+let menu = allEpisodes.forEach(element => {
+  let options = document.createElement('option');
+  options.innerHTML =  `${element.name}`
+  +  `-S${element.season.toString().padStart(2,'0')}E${element.number.toString().padStart(2,'0')}`;
+  myList.appendChild(options)
+});
+// episode selector/picker
+myList.addEventListener("change",function(w){
+let selector = w.target.value;
+let chosenEpi = allEpisodes.filter((key)=>{
+key =  `${key.name}`
++  `-S${key.season.toString().padStart(2,'0')}E${key.number.toString().padStart(2,'0')}`;
+return key === selector
+});
```

Comment:
Try to find an easier way to add functionality to the selector. It doesn't need to define a key again as because you defined an innerHtml to your option in the previous function.